pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1038,7 +1065,7 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 	if len(tls.CaCertificates) != 0 {
 		trustedCa = &core.DataSource{
 			Specifier: &core.DataSource_Filename{
-				Filename: model.GetOrDefault(metadata.TLSClientRootCert, tls.CaCertificates),
```

Comment:
add a check at line 1059. If tls == nil and service is internal service and autoMTLs is true, then do the transport socket matcher